### PR TITLE
Add build id to the image

### PR DESCRIPTION
**Pull request by david-caro**
This allows to see what version of the code is actually deployed in a
easy way (in the logs, and parameters).

### CODE REVIEW

File: pkg/server/validateadmission.go [hunk 1]
```diff
@@ -12,6 +12,13 @@ import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
+// patchOperation is an operation of a JSON patch, see https://tools.ietf.org/html/rfc6902 .
+type patchOperation struct {
+	Op    string      `json:"op"`
+	Path  string      `json:"path"`
+	Value interface{} `json:"value,omitempty"`
+}
+
```
Comment (david-caro) [inline pkg/server/validateadmission.go]: this is not needed

**Comment by blancadesal** [comment]
Basic question – what's the difference/advantage of adding the build id vs tagging the image with a git commit short hash?

**Comment by david-caro** [comment]
> Basic question – what's the difference/advantage of adding the build id vs tagging the image with a git commit short hash?

tl;dr; is that one tells you what the image you are pulling has, the other tells you about the "deployment" itself

Long version :)

There's several things that we might want to keep track of when deploying this service:
* which container image it is using (the specific build, and which commit was that generated with)
* which commit of the 'configuration' (helm charts/kustomize yamls/...) was used (which commit), and when

keep in mind that this change is not a full solution, just an easy step forward to one.

And we want to keep track of both differently as we might want to deploy the same container image, but using a different helm chart (ex. if we are reverting to a previous version).

This change only takes care of the second point up there, it allows to identify the commit of the config it was used to deploy, and when was it done.

There's some other changes to start using the git hash as container tag (see https://gerrit.wikimedia.org/r/c/operations/cookbooks/+/842351, and https://gerrit.wikimedia.org/r/c/labs/tools/registry-admission-webhook/+/841903) and a whole proposal to standardize/automate all of that (see https://wikitech.wikimedia.org/wiki/Wikimedia_Cloud_Services_team/EnhancementProposals/Toolforge_Kubernetes_component_workflow_improvements).

**Comment by blancadesal** [comment]
Thank you for taking the time to explain this, and to provide more context!

File: utils/realize_patch.sh [hunk 2]
```diff
@@ -1,14 +1,20 @@
...
-
-
 function main() {
     if [[ ${1:-} == '-v' ]]; then
         shift
         set -x
     fi
 
+    for template in $(find . -iname \*.tpl); do
+        realize_tempalte "$template"
```
Comment (dhinus) [inline utils/realize_patch.sh]: `s/tempalte/template/`

File: utils/realize_patch.sh [hunk 3]
```diff
@@ -26,8 +32,13 @@ function main() {
...
 
-    sed "s/@@CA_BUNDLE@@/${CA_BUNDLE}/g" "${template_path}" > "${template_path%.tpl}"
-    echo "Realized tmplate ${template_path} into ${template_path%.tpl}."
+    local git_hash="$(git rev-parse HEAD)"
+    sed \
+        -e "s/@@CA_BUNDLE@@/${CA_BUNDLE}/g" \
+        -e "s/@@BUILD_ID@@/${git_hash}-$(date +%Y%m%d_%H%M%S)/g" \
+        "${template_path}" \
+    > "${template_path%.tpl}"
+    echo "Realized tmplate ${template_path} into ${template_path%.tpl}"
```
Comment (dhinus) [inline utils/realize_patch.sh]: `s/tmplate/template/`

**Review by dhinus — CHANGES_REQUESTED** [review]
LGTM, but I found two typos :P

File: utils/realize_patch.sh (hunk 2 above)
Comment (david-caro) [reply utils/realize_patch.sh]: Ahahahah, and the autocomplete made sure it was consistent across the file! Xd

**Review by dhinus — APPROVED** [review]

**Comment by david-caro** [comment]
I'm pausing this until the deploy.sh one (#7) is in, as it changes some of the same files
